### ISSUE TITLE
Ephemeral Broker Storage Is Not Working 

### ISSUE THREAD

**Issue by geso02** [strimzi/strimzi-kafka-operator#6918]
**Describe the bug**
Ephemeral Storage seems not to work anymore in version 0.27.0 and 0.29.0

**To Reproduce**
Try this small resource

```
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: kafka-test
spec:
  kafka:
    version: 2.8.0
    replicas: 1
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
    storage: 
      type: ephemeral 
  zookeeper:
    replicas: 1
    storage:
      type: ephemeral
```

**Expected behavior**

There is no persistent-storage claim created.

** Behavior *

There is a PVC created for kafka. And the pod, waiting for it, does not start up.

**Environment :**
 - Strimzi version: 0.27.0 and 0.29.0 (worked with 0.26.1
 - Installation method: helm - strimzi-kafka-operator repo: https://strimzi.io/charts/
 - Kubernetes cluster: Kubernetes Major:"1", Minor:"23"
 - Infrastructure: [k3s cluster]

**Comment by scholzj** [comment]
Can you please provide the operator logs from the beginning of when you created the resource?

**Comment by geso02** [comment]
I recognized that it was not a problem with the version. I've started my setup previously with storage and used the newer version.
I deleted my helm installation and created it again but now with ephemeral storage, the PVC was still there ( but with now PC created and assigned ) and the pod was waiting for the PVC / PV to become available. Manually deletion of the PVC solved the problem. Seems to be more related to the kubernetes behavior, but I could not reproduce it anymore.
